tests/test_performance_patterns.py

```
@@ -0,0 +1,319 @@
+                event.sender = f"@user{i}:matrix.org"
+                event.server_timestamp = int(time.time() * 1000)
+
+                room = MagicMock()
+                room.room_id = mock_config["matrix_room_ids"][0]  # Use configured room
+                task = bot.on_room_message(room, event)
+                tasks.append(task)
+
+            await asyncio.gather(*tasks)
+
+            # Verify requests were processed
+            assert len(request_times) == 5
```

Ruff (0.12.2)

Use of `assert` detected

(S101)